c10/core/AutogradState.h

```diff
@@ -41,10 +50,15 @@ struct C10_API AutogradState {
     return inference_mode_;
   }
 
+  bool get_multithreading_enabled() const {
+    return mulithreading_enabled_;
+  }
+
  private:
   bool grad_mode_ : 1;
   bool inference_mode_ : 1;
   bool fw_grad_mode_ : 1;
+  bool mulithreading_enabled_ : 1;
```

Comment:
I'm curious if you really want this to be thread local. 
Aren't some of the things you want to catch is when you end up in another thread but the ThreadLocalState wasn't propagated?

Reply:
In the context of torch deploy, IMO it makes more sense for this to be thread local. And, if this is actually being respected in the backwards, we shouldn't have the problem of copying over TLS. But open to suggestions

Reply:
The TLS access shouldn't be too much of a perf regression here. So that's fine by me. Just wanted to double check.